src/Data/Functor/Contravariant/Conclude.hs

```diff
@@ -0,0 +1,185 @@
+{-# LANGUAGE CPP           #-}
+{-# LANGUAGE TypeOperators #-}
+
+-----------------------------------------------------------------------------
+-- |
+-- Copyright   :  (C) 2011-2015 Edward Kmett
+-- License     :  BSD-style (see the file LICENSE)
+--
+-- Maintainer  :  Edward Kmett <ekmett@gmail.com>
+-- Stability   :  provisional
+-- Portability :  portable
+--
+----------------------------------------------------------------------------
+module Data.Functor.Contravariant.Conclude (
+    Conclude(..)
+  , concluded
+  ) where
+
+import Control.Applicative.Backwards
+import Control.Monad.Trans.Identity
+import Control.Monad.Trans.List
+import Control.Monad.Trans.Maybe
+import qualified Control.Monad.Trans.RWS.Lazy as Lazy
+import qualified Control.Monad.Trans.RWS.Strict as Strict
+import Control.Monad.Trans.Reader
+import qualified Control.Monad.Trans.State.Lazy as Lazy
+import qualified Control.Monad.Trans.State.Strict as Strict
+import qualified Control.Monad.Trans.Writer.Lazy as Lazy
+import qualified Control.Monad.Trans.Writer.Strict as Strict
+
+import Data.Functor.Apply
+import Data.Functor.Compose
+import Data.Functor.Contravariant
+import Data.Functor.Contravariant.Decide
+import Data.Functor.Contravariant.Divise
+import Data.Functor.Contravariant.Divisible
+import Data.Functor.Product
+import Data.Functor.Reverse
+import Data.Void
+
+#if !MIN_VERSION_base(4,8,0)
+import Control.Applicative
+#endif
+
+#if MIN_VERSION_base(4,8,0)
+import Data.Monoid (Alt(..))
+#else
+import Data.Monoid (Monoid(..))
+#endif
+
+#if MIN_VERSION_base(4,7,0) || defined(MIN_VERSION_tagged)
+import Data.Proxy
+#endif
+
+#ifdef MIN_VERSION_StateVar
+import Data.StateVar
+#endif
+
+#if __GLASGOW_HASKELL__ >= 702
+#define GHC_GENERICS
+import GHC.Generics
+#endif
+
+-- | The contravariant analogue of 'Plus'.  Adds on to 'Decide' the ability
+-- to express a combinator that rejects all input, to act as the dead-end.
+-- Essentially 'Decidable' without a superclass constraint on 'Divisible'.
+--
+-- If one thinks of @f a@ as a consumer of @a@s, then 'conclude' defines
+-- a consumer that cannot ever receive /any/ input.
+--
+-- Conclude acts as an identity with 'decide', because any decision that
+-- involves 'conclude' must necessarily /always/ pick the other option.
+--
+-- That is, for, say,
+--
+-- @
+-- 'decide' f x 'concluded'
+-- @
+--
+-- @f@ is the deciding function that picks which of the inputs of @decide@
+-- to direct input to; in the situation above, @f@ must /always/ direct all
+-- input to @x@, and never 'concluded'.
+--
+-- Mathematically, a functor being an instance of 'Decide' means that it is
+-- "monoidal" with respect to the contravariant "either-based" Day
```

Comment:
Haddock will render things between double quotes as module names, so make sure to escape them here:

```suggestion
-- \"monoidal\" with respect to the contravariant \"either-based\" Day
```